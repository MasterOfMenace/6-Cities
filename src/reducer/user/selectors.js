import NameSpace from '../name-space.js';

export const getAuthorizationStatus = (state) => {
  return state[NameSpace.USER].authorizationStatus;
};

export const getUserInfo = (state) => {
  return state[NameSpace.USER].userInfo;
};
