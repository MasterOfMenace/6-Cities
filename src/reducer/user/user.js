import {Adapter} from '../../adapter/adapter.js';
import {ActionCreator as AppActionCreator} from '../app-reducer/app-reducer.js';
import {Message} from '../../const.js';

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

const ResponseStatus = {
  OK: 200
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  userInfo: null
};

export const ActionType = {
  REQUIRE_AUTHORIZATION: `require_authorization`,
  GET_USER_INFO: `get_user_info`
};

export const ActionCreator = {
  requireAuthorization: (status) => ({
    type: ActionType.REQUIRE_AUTHORIZATION,
    payload: status
  }),

  getUserInfo: (authData) => ({
    type: ActionType.GET_USER_INFO,
    payload: authData
  })
};

export const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`login`)
      .then((response) => {
        if (response.status === ResponseStatus.OK) {
          const userInfo = Adapter.getUserInfo(response.data);
          dispatch(ActionCreator.getUserInfo(userInfo));
        }
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      })
      .catch((err) => {
        const code = err.response.status;
        dispatch(AppActionCreator.changeErrMessage(Message[code]));
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`login`, {
      email: authData.login,
      password: authData.password,
    })
      .then((response) => {
        const userInfo = Adapter.getUserInfo(response.data);
        dispatch(ActionCreator.getUserInfo(userInfo));
      })
      .then(() => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      });
  }
};

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRE_AUTHORIZATION:
      return Object.assign({}, state, {
        authorizationStatus: action.payload
      });

    case ActionType.GET_USER_INFO:
      return Object.assign({}, state, {
        userInfo: action.payload
      });
  }

  return state;
};
