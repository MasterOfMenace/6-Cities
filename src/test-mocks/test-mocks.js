export const cities = [
  {
    name: `Amsterdam`,
    location: [52.38333, 4.9],
    zoom: 12
  },
  {
    name: `Paris`,
    location: [48.85657, 2.35186],
    zoom: 12
  },
  {
    name: `Cologne`,
    location: [50.93685, 6.9625],
    zoom: 12
  },
  {
    name: `Brussels`,
    location: [50.85300, 4.43792],
    zoom: 12
  },
  {
    name: `Hamburg`,
    location: [53.53539, 9.99371],
    zoom: 12
  },
  {
    name: `Dusseldorf`,
    location: [51.22977, 6.79138],
    zoom: 12
  },
];

export const offers = [
  {
    id: 1,
    city: {
      name: cities[0].name
    },
    title: `Offer1`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 100,
    rating: 4,
    type: `Private room`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
  {
    id: 2,
    city: {
      name: cities[1].name
    },
    title: `Offer2`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 200,
    rating: 4,
    type: `Apartment`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
  {
    id: 3,
    city: {
      name: cities[2].name
    },
    title: `Offer3`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 300,
    rating: 4,
    type: `Hostel`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
  {
    id: 4,
    city: {
      name: cities[3].name
    },
    title: `Offer4`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 400,
    rating: 4,
    type: `Apartment`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
  {
    id: 5,
    city: {
      name: cities[4].name
    },
    title: `Offer1`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 100,
    rating: 4,
    type: `Private room`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
  {
    id: 6,
    city: {
      name: cities[5].name
    },
    title: `Offer2`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 200,
    rating: 4,
    type: `Apartment`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
  {
    id: 7,
    city: {
      name: cities[1].name
    },
    title: `Offer3`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 300,
    rating: 4,
    type: `Hostel`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
  {
    id: 8,
    city: {
      name: cities[2].name
    },
    title: `Offer4`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 400,
    rating: 4,
    type: `Apartment`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
];

export const reviews = [
  {
    author: {
      avatar: `author-avatar-1`,
      id: 1,
      isPro: false,
      name: `author-1`
    },
    id: 1,
    rating: 3,
    text: `Review text 1`,
    date: `date`
  },
  {
    author: {
      avatar: `author-avatar-2`,
      id: 2,
      isPro: false,
      name: `author-2`
    },
    id: 2,
    rating: 5,
    text: `Review text 2`,
    date: `date`
  }
];

export const neighbors = [
  {
    id: 4,
    city: {
      name: `City`,
      location: [52.38333, 4.9]
    },
    title: `Offer4`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 400,
    rating: 4,
    type: `Apartment`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  }
];

export const userInfo = {
  avatarUrl: `/img/1.png`,
  email: `Oliver.conner@gmail.com`,
  id: 1,
  isPro: false,
  name: `Oliver.conner`
};
