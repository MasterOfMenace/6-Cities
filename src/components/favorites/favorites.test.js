import React from 'react';
import {Router} from 'react-router-dom';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import Favorites from './favorites.jsx';
import history from '../../history.js';
import {AuthorizationStatus} from '../../reducer/user/user.js';
import {offers as mockOffers, userInfo as mockUserInfo} from '../../test-mocks/test-mocks.js';
import NameSpace from '../../reducer/name-space.js';

const mockStore = configureStore([]);

describe(`Правильное отображение компонента Favorites`, () => {
  it(`Без избранных предложений`, () => {

    const store = mockStore({
      [NameSpace.DATA]: {
        favorites: [],
      },
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
        userInfo: mockUserInfo
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <Router history={history}>
            <Favorites
              authorizationStatus={AuthorizationStatus.AUTH}
              userInfo={mockUserInfo}
              favoriteOffers={[]}
              onMouseLeave={()=>{}}
              onMouseOver={()=>{}}
              onFavoriteButtonClick={()=>{}}
            />
          </Router>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`С избранными предложениями`, () => {

    const store = mockStore({
      [NameSpace.DATA]: {
        favorites: mockOffers,
      },
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
        userInfo: mockUserInfo
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <Router history={history}>
            <Favorites
              authorizationStatus={AuthorizationStatus.AUTH}
              userInfo={mockUserInfo}
              favoriteOffers={mockOffers}
              onMouseLeave={()=>{}}
              onMouseOver={()=>{}}
              onFavoriteButtonClick={()=>{}}
            />
          </Router>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
