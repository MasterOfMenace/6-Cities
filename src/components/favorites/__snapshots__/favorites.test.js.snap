// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Правильное отображение компонента Favorites Без избранных предложений 1`] = `
<div
  className="page page--favorites-empty"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="User avatar"
                    className="user__avatar"
                    height="20"
                    src="/img/1.png"
                    width="20"
                  />
                </div>
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites-empty"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites favorites--empty"
      >
        <h1
          className="visually-hidden"
        >
          Favorites (empty)
        </h1>
        <div
          className="favorites__status-wrapper"
        >
          <b
            className="favorites__status"
          >
            Nothing yet saved.
          </b>
          <p
            className="favorites__status-description"
          >
            Save properties to narrow down search or plan yor future trips.
          </p>
        </div>
      </section>
    </div>
  </main>
  <footer
    className="footer"
  >
    <a
      className="footer__logo-link"
      href="main.html"
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;

exports[`Правильное отображение компонента Favorites С избранными предложениями 1`] = `
<div
  className="page null"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="User avatar"
                    className="user__avatar"
                    height="20"
                    src="/img/1.png"
                    width="20"
                  />
                </div>
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favoritesnull"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Amsterdam
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={110}
                      src="img/apartment-01.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        100
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button   button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/1"
                      onClick={[Function]}
                    >
                      Offer1
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  />
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Paris
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={110}
                      src="img/apartment-01.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        200
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button   button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/2"
                      onClick={[Function]}
                    >
                      Offer2
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  />
                </div>
              </article>
              <article
                className="favorites__card place-card"
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={110}
                      src="img/apartment-01.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        300
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button   button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/7"
                      onClick={[Function]}
                    >
                      Offer3
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  />
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Cologne
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={110}
                      src="img/apartment-01.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        300
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button   button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/3"
                      onClick={[Function]}
                    >
                      Offer3
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  />
                </div>
              </article>
              <article
                className="favorites__card place-card"
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={110}
                      src="img/apartment-01.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        400
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button   button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/8"
                      onClick={[Function]}
                    >
                      Offer4
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  />
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Brussels
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={110}
                      src="img/apartment-01.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        400
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button   button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/4"
                      onClick={[Function]}
                    >
                      Offer4
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  />
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Hamburg
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={110}
                      src="img/apartment-01.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        100
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button   button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/5"
                      onClick={[Function]}
                    >
                      Offer1
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  />
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Dusseldorf
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={110}
                      src="img/apartment-01.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        200
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button   button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/6"
                      onClick={[Function]}
                    >
                      Offer2
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  />
                </div>
              </article>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer"
  >
    <a
      className="footer__logo-link"
      href="main.html"
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
